colossalai/zero/init_ctx/init_context.py

```diff
@@ -242,6 +242,10 @@ def half_fn(t: torch.Tensor):
             # mark whether the param is replicated
             param.is_replicated = self.is_replicated
```

Comment:
As other attributes are set in `param.colo_attr`. Should we also set `is_replicated` and `keep_not_shard` in `param.colo_attr`?

Reply:
Since `colo_attr` is an instance of `ShardedParamV2`, which is assumed to be sharded naturally.
It is weird to add an attribute `keep_not_shard` for it.

Reply:
done